webapp/graphql/User.js

```diff
@@ -258,3 +258,24 @@ export const checkSlugAvailableQuery = gql`
     }
   }
 `
+
+export const currentUserQuery = gql`
+  ${userFragment}
+  query {
+    currentUser {
+      ...user
+      email
+      role
+      about
+      locationName
+      locale
+      allowEmbedIframes
```

Comment:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-04T12:53:00Z" title="Wednesday, March 4th 2020, 1:53:00 pm +01:00">Mar 4, 2020</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -258,3 +258,24 @@ export const checkSlugAvailableQuery = gql`
     }
   }
 `
+
+export const currentUserQuery = gql`
+  ${userFragment}
+  query {
+    currentUser {
+      ...user
+      email
+      role
+      about
+      locationName
+      locale
+      allowEmbedIframes
```


I miss  `contributionsCount` and `commentedCount`

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-03-04T13:00:08Z" title="Wednesday, March 4th 2020, 2:00:08 pm +01:00">Mar 4, 2020</time>_

---


That's on purpose. Fetching the two will cause unnecessary N+1 queries to the database. It seems that we don't use the counts in the frontend.

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-04T13:08:56Z" title="Wednesday, March 4th 2020, 2:08:56 pm +01:00">Mar 4, 2020</time>_

---


And the profile page?

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-04T13:19:29Z" title="Wednesday, March 4th 2020, 2:19:29 pm +01:00">Mar 4, 2020</time>_

---


see webapp/pages/profile/_id/_slug.vue line 179 and 191
`<hc-count-to slot="count" :end-val="user.contributionsCount" />`
`<hc-count-to slot="count" :end-val="user.commentedCount" />`

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2020-03-04T13:40:35Z" title="Wednesday, March 4th 2020, 2:40:35 pm +01:00">Mar 4, 2020</time>_

---


The `user` is not coming from `store/auth` but from a separate graphql query: https://github.com/Human-Connection/Human-Connection/blob/master/webapp/pages/profile/_id/_slug.vue#L507

So that's no problem. The `currentUser` is what you see when you go to your avatar menu and update e.g. your social media.

Reply:
> [<img alt="Mogge" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Mogge) **Authored by [Mogge](https://github.com/Mogge)**
_<time datetime="2020-03-04T13:57:22Z" title="Wednesday, March 4th 2020, 2:57:22 pm +01:00">Mar 4, 2020</time>_

---


OK. Got it. In the profile we call the UserQuery where the counts are included. So you are right. It is not needed here.